cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/SystemInformation.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2025, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.frontends.python
+
+import de.fraunhofer.aisec.cpg.graph.OverlayNode
+
+/** Represents the contents of `sys.version_info` which contains the Python version. */
+data class VersionInfo(var major: Long? = null, var minor: Long? = null, var micro: Long? = null) :
+    OverlayNode() {
+    /**
+     * Returns the version info as a tuple (major, minor, micro). The length of the tuple depends on
+     * the information set, e.g., if only major version is set, then the tuple is 1 element long.
+     */
+    fun toList(): List<Long> {
+        val list = mutableListOf<Long>()
+        major?.let { major ->
+            list += major
+            minor?.let { minor ->
+                list += minor
+                micro?.let { micro -> list += micro }
+            }
+        }
+
+        return list
+    }
+}
+
+/**
+ * Represents different system information that are used in the [PythonValueEvaluator] to evaluate
+ * expressions, such as `sys.platform` and `sys.version_info`.
+ *
+ * We model this as an overlay node so we can access this information later in the graph.
+ */
+data class SystemInformation(var versionInfo: VersionInfo? = null, var platform: String? = null) :
```

Comment:
This is not necessarily python specific but describes something like the runtime system of a program, so I'd say it can be moved to `cpg-concept`. Same for `VersionInfo` (maybe call it `RuntimeVersion` or so to not confuse it with the version of the program we analyze?)

Reply:
We may move these classes to `cpg-core` even if this also has drawbacks. But for now, I'd also be fine keeping them in python only